src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -135,6 +139,69 @@ public Task WriteTextAsync(SourceText text, CancellationToken cancellationToken
                     // See commentary in ReadTextAsync for why this is implemented this way.
                     return Task.Factory.StartNew(() => WriteText(text, cancellationToken), cancellationToken, TaskCreationOptions.None, TaskScheduler.Default);
                 }
+
+                private unsafe TextReader CreateTextReaderFromTemporaryStorage(ISupportDirectMemoryAccess accessor, int streamLength, CancellationToken cancellationToken)
+                {
+                    char* src = (char*)accessor.GetPointer();
+
+                    // BOM: Unicode, little endian
+                    // Skip the BOM when creating the reader
+                    Debug.Assert(*src == 0xFEFF);
```

Comment:
because we are the one who write in and read it and no-one else has access to it.